client/src/lua/config.rs

```diff
@@ -28,13 +29,16 @@ pub fn load_config(lua: Context, cmdline_config: Option<&str>) -> io::Result<(St
             .into_os_string()
             .into_string()
             .expect("init_dir not a valid UTF-8 string");
+        trace!("Adding location of init file to package.path: {}", init_dir);
         package
             .set("path", format!("{};{}", paths, init_dir))
             .expect("Failed to set package.path");
     }
     let mut init_contents = String::new();
     init_file.read_to_string(&mut init_contents)?;
-    Ok(("init.lua".to_string(), init_contents))
+    let file_name = init_path.file_name().map(OsStr::to_str).unwrap_or(Option::None);
```

Comment:
Use `and_then` instead of `map(...).unwrap_or(Option::None)`.

[Option.and_then](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then)

This is also called "flat map" in other languages (haskell? OCaml? At least one of those I think)

Reply:
Thanks, I knew there had to be a function that does this but must have overlooked it..